### ISSUE TITLE
Improve binary focal loss to trade off recall and precision by adding weights to positive examples like torch.nn.BCEWithLogitsLoss

### ISSUE THREAD

**Issue by xiaoyuan0203** [kornia/kornia#1744]
## 🚀 Feature
Improve binary focal loss to trade off recall and precision by adding weights to positive examples. Just like `pos_weight` parameter in torch.nn.BCEWithLogitsLoss. `pos_weight` must be a vector with length equal to the number of classes.

**Comment by edgarriba** [comment]
Open to send a PR ?

**Comment by MrShevan** [comment]
@xiaoyuan0203 @edgarriba Looks interesting! Can I take it?

**Comment by edgarriba** [comment]
@MrShevan sure ! go ahead

**Comment by xiaoyuan0203** [comment]
#> @xiaoyuan0203 Open to send a PR ?

Sorry for reply to you until now. Could you tell me how to do it?

**Comment by MrShevan** [comment]
Oh sorry. I see you are interested in creating your own PR. Let me know if it's free

**Comment by xiaoyuan0203** [comment]
It's free of cource. It's my first time to take part in open-source,so I just want to know how to do. If you are interested in make a pull request, you can do it. Do you mean you want to improve the focal loss?

**Comment by edgarriba** [comment]
@xiaoyuan0203 i have other suggestions for losses to add e.g for semantic segmentation. or the ones from this model we are going to include: https://github.com/xavysp/LDC

**Comment by MrShevan** [comment]
@xiaoyuan0203 Well, I can help you get started :)
We can try to make this PR together. You can fork this and create your own branch. Here it is well written in paragraph `Prepare your Workspace`. And then I'll join you)

**Comment by edgarriba** [comment]
you can also check our less graphical contributor guides: https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst#developing-kornia

**Comment by xiaoyuan0203** [comment]
@MrShevan Here? https://github.com/xiaoyuan0203/kornia/tree/feat/focal_loss

**Comment by MrShevan** [comment]
Yep, great)
I have done the following steps:
1. I cloned your forked repo into my local and switched into your branch. 
I use PyCharm as IDE, so it was simple to do the initial setup
<img width="1440" alt="Снимок экрана 2022-06-30 в 10 15 15" src="https://user-images.githubusercontent.com/32237302/176621692-0596d00a-e3dc-41fb-9343-a4f48e11e295.png">

and switch to `feat/focal_loss`
<img width="373" alt="Снимок экрана 2022-06-30 в 10 33 33" src="https://user-images.githubusercontent.com/32237302/176621918-9ccb9a2f-7be2-4df4-80d3-1c2d9024e720.png">

2. The next step is setting your python interpreter and run all tests **before starting coding**. :)

I ran all commands from [this guide](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst#developing-kornia) in 4:
```
source ./path.bash.inc
python setup.py develop
python -c "import kornia; print(kornia.__version__)"
```
And then install all requirements to install pytest and etc.:
```
pip install --ignore-installed -r requirements/dev.txt
```

Now I can run all related to our PR tests (cause it is important to make sure that all the tests that we will change worked initially):
<img width="1440" alt="Снимок экрана 2022-06-30 в 10 19 05" src="https://user-images.githubusercontent.com/32237302/176623161-bb2e9687-a9ee-44ab-855c-3d44866fa321.png">

And they all passed :)
<img width="1418" alt="Снимок экрана 2022-06-30 в 10 52 32" src="https://user-images.githubusercontent.com/32237302/176623504-5107f2e5-b0bf-4cc2-a7ec-670878cedb3b.png">

Well, now I'm ready for coding

**Comment by xiaoyuan0203** [comment]
@MrShevan Thanks a lot! Is it a good habit to use my usual conda environment instead of environment from the `kornia/setup.py`?

**Comment by MrShevan** [comment]
Well, I think If the tests run, it doesn't matter. But sometimes tests too complicated and require a lot of installed python libraries, in such case easy create new interpreter as I did or updateConda (command from [this guide](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst#developing-kornia) in 4: `./setup_dev_env.sh`)

Do you mind if I open the first PR from your branch with my draft changes?)

**Comment by xiaoyuan0203** [comment]
It doesn't matter.
